File: 633_sum_of_square_numbers/judgeSquareSum.ts
function judgeSquareSum(c: number): boolean {
  let start = 0;
  let end = Math.floor(Math.sqrt(c));

  while (start <= end) {
    const powSum = Math.pow(start, 2) + Math.pow(end, 2);

    if (powSum === c) {
      return true;
    } else if (powSum < c) {
      start++;
    } else {
      end--;
    }
  }

  return false;
}

File: 633_sum_of_square_numbers/judgeSquareSum.py
class Solution:
    def judgeSquareSum(self, c: int) -> bool:
        start = 0
        end = int(sqrt(c))
        
        while start <= end:
            powSum = pow(start, 2) + pow(end, 2)

            if powSum == c:
                return True
            elif powSum > c:
                end -= 1
            else:
                start += 1

        return False